### DISCUSSION TITLE
clicking on multiple images

### DISCUSSION THREAD

**Discussion by veranicolette** [jspsych/jsPsych#1219] [Q&A]
Hi, I'm creating a bot checker for my experiment in which the participant must click on the correct images in order to proceed to the main task. I used the plugin, html button response and I already have the images set up with the prompt at the bottom. Is there a way I can allow the user to click on multiple images and show a "success" page if they click on all the right images?

**Comment by jodeleeuw** [comment]
Hi @veranicolette ,

You'll need to modify the plugin to handle this. If you want a procedure that's compatible with the existing plugins you could show a sequence of trials with a single target image among distractors, and then at the end of the sequence you can show a success of failure message. You can generate the message at the end using [dynamic parameters](https://www.jspsych.org/overview/trial/#dynamic-parameters). Something like the following:

```js
var feedback_message = {
	type: 'html-keyboard-response',
	stimulus: function(){
		var all_correct = jsPsych.data.get().filter({task: 'bot-detection'}).select('correct').all(function(x) { return x==true});
		if(all_correct){
			return '<p>You are not a bot!</p>'
		} else {
			return '<p>BOT!</p>
		}
	}
}
```

This will require using the `data` parameter to label each image detection trial:

```js
data: {task: 'bot-detection'}
```

And you'll need to create a `correct` variable in the `on_finish` function of the image detection trials too:

```js
on_finish: function(data){
	// check if the response is correct, and if it is set
	data.correct = true
}
```
